code/modules/vending/_vending.dm

```diff
@@ -187,6 +196,7 @@ GLOBAL_LIST_EMPTY(vending_products)
 		R.max_amount = amount
 		R.custom_price = initial(temp.custom_price)
 		R.custom_premium_price = initial(temp.custom_premium_price)
+		R.age_restricted = initial(temp.age_restricted)
```

Comment:
If you move this var out of /atom (and you REALLY REALLY SHOULD) you'll need to change this.
If, for example, you moved it to /obj/item:
```dm
if (istype(temp, /obj/item))
    var/obj/item/O = temp
    R.age_restricted = initial(O.age_restricted)
else
    R.age_restricted = FALSE
```